dynamo-test/src/test/scala/d4s/DynamoInterpreterTest.scala

```diff
@@ -215,18 +211,15 @@ final class DynamoInterpreterTest extends DynamoTestBase[Ctx] with DynamoRnd {
       ctx =>
         import ctx._
         import ctx.testTable.table
-        import io.circe.{Codec, derivation}
 
         final case class AdditionalFields(field4: String, field5: Int)
         object AdditionalFields {
-          implicit val codec: Codec.AsObject[AdditionalFields] = derivation.deriveCodec[AdditionalFields]
+          implicit val codec: D4SCodec[AdditionalFields] = D4SCodec.derive[AdditionalFields]
         }
 
         final case class ExtendedPayload(payload: InterpreterTestPayload, additionalFields: AdditionalFields)
         object ExtendedPayload {
-          implicit val decoder: Decoder[ExtendedPayload] = {
-            Decoder[InterpreterTestPayload].map2(Decoder[AdditionalFields])(ExtendedPayload(_, _))
-          }
+          implicit val codec: D4SCodec[ExtendedPayload] = D4SCodec.derive
```

Comment:
@neko-kai, I've changed a little bit the test here, because apparently I couldn't combine D4SCodec instances using  apply syntax `Decoder[InterpreterTestPayload].map2(Decoder[AdditionalFields])(ExtendedPayload(_, _))`
Probably need to provide Semigroup instance for `D4SDecoder/Encoder` in future.